image-classifier/app.R

```diff
@@ -2,89 +2,110 @@ library(magrittr)
 library(reticulate)  # Used to call Tensorflow Python script
 library(shiny)
 library(shinycssloaders)
+library(gitlink)
+
+behavior <- config::get("image")
+stopifnot(behavior %in% c("upload", "download"))
 
 # Load source of Python image classifier script
 source_python('image-classifier.py')
 
 server <- function(input, output, session) {
-
-    output$contents <- renderTable({
-        # After the user uploads a file, the image will be classified and the predictions will be shown.
+    
+    # where the image that should be classified is on disk
+    image_path <- reactiveVal("./img/cat.jpg")
+    
+    image_prefix <- "pytorch_image"
+    
+    # the configurable selector for download vs. upload
+    output$image_selector <- renderUI({
+        if (behavior == "download") {
+          list(
+            textInput("file1", label = h5("Enter Image URL:"), value = ""),
+            actionButton("download1", "Download")
```

Comment:
I think this would be more clear as "Fetch Image" or something else, since "Download" makes me think that it is going to download an image to my local machine.

Reply:
I'd prefer no spaces / lowercase - is `fetch-image` ok?

Reply:
Sorry, I meat the higher level term that is used to refer to this action, not that specific occurrence.

I think "download" in all occurrences in this file should be referred to as something else to avoid confusion, including the UI buttons/action, since it is not actually downloading anything from the UI to the user's machine.

In other words, we are using "download" to mean from the internet to the app, but "upload" to mean from the user's machine to the app.

Reply:
I think that's a good call, and that `fetch-image-url` makes sense as both a config option and in the code. The one downside of `-` is that it is not valid R syntax, so using `input$` syntax requires backticks... is that enough to push us to `fetch_image_url`?

Reply:
Yep, `fetch_image_url` is good. Or just `fetch` if you want to go minimal.